### PR TITLE
[WIP] mypy: Add /tools to test mypy annotations.

### PR DESCRIPTION
**Pull request by Shivamgera**
It tests the mypy annotations in the code inside ```zulipterminal``` directory,  that is, whenever the code is modified or not, we can check if it is mypy annotated or not by running ```./tools/run-mypy```.

Currently, it has three optional arguments:

- ```--modified```: To check all modified files.
- ```--all```: To check all the files.
- ```--quick```: To use the quick flag with mypy.

@amanagr, can you please review it?

### CODE REVIEW

**Comment by amanagr** [comment]
@Shivamgera, Thanks for working on this.
You must not mix spaces and tabs in your code. **ONLY USE SPACES**.

**Comment by amanagr** [comment]
@Shivamgera, have you looked at the error messages?
It shows `error: Cannot find module named 'urwid'`, can you resolve this?

**Comment by timabbott** [comment]
@Shivamgera your PR description should have made clear what you did to generate this PR.

**Comment by Shivamgera** [comment]
@amanagr, It seems rebasing created some problems for me. Shall I close this PR and open a new one?

File: tools/run-mypy [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+sys.path.append(os.path.dirname(TOOLS_DIR))
+
+parser = argparse.ArgumentParser(description="Run mypy on files tracked by git.")
+parser.add_argument('targets',nargs="*", default=[],help="""files and directories to include in the result.
+                    If this is not specified, the current directory is used""")
+parser.add_argument('-m','--modified', action = 'store_true', help = 'All modified files.',default = False)
+parser.add_argument('-a','--all', action = 'store_true')
+parser.add_argument('--quick', action='store_true', default=False,
+                    help="Use the quick flag with mypy" )
+parser.add_argument('--ignore-missing-imports', dest='ignore_missing_imports', action='store_true', default=False,
```
Comment (amanagr) [inline tools/run-mypy]: We don't need this.

**Comment by amanagr** [comment]
@**Shivam Gera**, Can you do this: 
* copy the code from https://github.com/zulip/python-zulip-api/blob/master/tools/run-mypy
* remove `__future__` imports
* remove exclude/incude vars

**Comment by Shivamgera** [comment]
@amanagr, Done.

**Comment by amanagr** [comment]
@Shivamgera, I merged it after some changes. Thanks!

**Pull request closed without merging**